packages/nuxt/src/components/runtime/server-component.ts

```diff
@@ -46,17 +48,19 @@ export const createIslandPage = (name: string) => {
       expose({
         refresh: () => islandRef.value?.refresh(),
       })
-
+      const nuxtApp = useNuxtApp()
       const route = useRoute()
       const path = import.meta.client && await isPrerendered(route.path) ? route.path : route.fullPath.replace(/#.*$/, '')
-
       return () => {
         return h('div', [
           h(NuxtIsland, {
             name: `page:${name}`,
             lazy: props.lazy,
             ref: islandRef,
             context: { url: path },
+            onError: (e) => {
+              nuxtApp.runWithContext(() => showError(e))
```

Comment:
should we do this only on the client-side?

Reply:
we need to do it on both sides, otherwise a blank page will show up

Reply:
and just to be clear, it doesn't render an error page within the 'island' slot - it fully shows the error, right?

Reply:
Yes, do you prefer to throw the error server side instead ?